test/ecto/adapters/postgres_test.exs

```diff
@@ -395,6 +405,19 @@ defmodule Ecto.Adapters.PostgresTest do
     assert all(query) == String.trim(result)
   end
 
+  test "order_by and types" do
+    query =
+      normalize from(e in "schema4",
+        order_by: type(fragment("(?->>?)", field(e, ^:map), ^"value"), ^:decimal),
+        select: true)
+
+    result =
+      "SELECT TRUE FROM \"schema4\" AS s0 " <>
+      "ORDER BY (s0.\"map\"->>$1)::decimal"
+
+    assert all(query) == String.trim(result)
```

Comment:
You don't need trim here. YOu can keep everything in a single line:

    assert all(query) ==  "SELECT TRUE FROM \"schema4\" AS s0 ORDER BY (s0.\"map\"->>$1)::decimal"